opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -387,7 +387,8 @@ def values(self) -> typing.ValuesView[str]:
 
 
 DEFAULT_TRACE_STATE = TraceState.get_default()
-_TRACE_ID_HEX_LENGTH = 2 ** 128 - 1
+_TRACE_ID_MAX_VALUE = 2 ** 128 - 1
+_SPAN_ID_MAX_VALUE = 2 ** 64 - 1
```

Comment:
Does this hold true for custom id generators?

Reply:
I think this should be fine, the [OTLP proto semantics](https://github.com/open-telemetry/opentelemetry-proto/blob/60fa8754d890b5c55949a8c68dcfd7ab5c2395df/opentelemetry/proto/trace/v1/trace.proto#L77) say it should be an 8-byte array.

Reply:
[Tracing API spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spancontext) also defines with a length of 8 bytes.

Additionally, some exporters will fail with an `OverflowError` when trying to convert a larger/negative value to a protobuf representation: e.g. the OTLP [HTTP](https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/trace_exporter/encoder/__init__.py#L298) or [GRPC](https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/trace_exporter/__init__.py#L133) exporters.